types/react/index.d.ts

```diff
@@ -278,7 +278,7 @@ declare namespace React {
 
     // Base component for plain JS classes
     // tslint:disable-next-line:no-empty-interface
-    interface Component<P = {}, S = {}> extends ComponentLifecycle<P, S> { }
+    interface Component<P = {}, S = {}, SS = never> extends ComponentLifecycle<P, S, SS> { }
```

Comment:
Clarifying question: with this change would the expectation be that any new component created, that uses snapshots, would need to specify the snapshot type in the class declaration?

Reply:
That's the idea; otherwise this would have had to be typed as `any`.